### PR TITLE
SHS-5808: Uninstall Webform module

### PR DESCRIPTION
**Pull request by codechefmarc**
# READY FOR REVIEW

## Summary
- Removes config for `hs_webform` module, paragraphs, and related field config
- Removes `hs_webform` dependencies from config ignored configs
- Uninstalls the `hs_webform` module
- Removes config for `webform` and `webform_ui` modules
- Uninstalls `webform` and `webform_ui` modules
- Does not remove `hs_webform` or `webform` from the filesystem - to be done in a subsequent PR

## Need Review By (Date)
2024-09-23

## Urgency
low

## Steps to Test
1. Login to the site as an admin
2. Visit the paragraph types page `/admin/structure/paragraphs_type` and verify that there is no longer a "Webform" paragraph type (`hs_webform`)
3. Visit the modules listing page `/admin/modules` and verify that the following modules are now uninstalled: `hs_webform`, `webform_ui` and `webform`.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208357464145158

### CODE REVIEW

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-colorful-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-traditional-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0

Visual Diffs:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-colorful-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-traditional-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0

Visual Diffs:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-colorful-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-traditional-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0

Visual Diffs:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0/visualdiffs

File: config/default/user.role.authenticated.yml [hunk 1]
```diff
@@ -27,8 +26,8 @@ permissions:
   - 'access shortcuts'
   - 'delete own files'
   - 'download media'
+  - 'opt-in or out of google analytics tracking'
```
Comment (cienvaras) [inline config/default/user.role.authenticated.yml]: Please remove this line. This config is ignored, but it's better to avoid adding unrelated changes.

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install [hunk 2]
```diff
@@ -923,3 +927,111 @@ function su_humsci_profile_update_9707() {
...
+  foreach ($config_names as $config) {
+    $config_factory->getEditable($config)->delete();
+    $message = t('Deleted @config', [
+      '@config' => $config,
+    ]);
+    Drush::logger()->notice($message);
+  }
+
+  // Uninstalls modules after removing configs to prevent updb errors.
+  \Drupal::service('module_installer')->uninstall(['webform_ui', 'webform']);
```
Comment (cienvaras) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Shouldn't `hs_webform` module also be uninstalled here?

File: config/default/field.field.node.hs_basic_page.field_hs_page_components.yml [hunk 3]
```diff
@@ -31,78 +30,74 @@ settings:
...
+        weight: -26
         enabled: false
       hs_timeline_item:
         weight: -27
         enabled: true
       hs_view:
-        weight: -31
+        weight: -25
         enabled: false
-      hs_webform:
```
Comment (cienvaras) [inline config/default/field.field.node.hs_basic_page.field_hs_page_components.yml:97]: In overridden config, try to include only the lines necessary for the the `hs_webform` paragraph removal. All other changes might be different from site to site and it's better to leave them out of the PR.

File: config/default/field.field.node.hs_basic_page.field_hs_page_hero.yml [hunk 4]
```diff
@@ -28,77 +28,71 @@ settings:
...
+        weight: -27
         enabled: false
       hs_timeline_item:
-        weight: -28
+        weight: -26
         enabled: false
       hs_view:
-        weight: -34
-        enabled: false
-      hs_webform:
```
Comment (cienvaras) [inline config/default/field.field.node.hs_basic_page.field_hs_page_hero.yml:95]: In overridden config, try to include only the lines necessary for the the `hs_webform` paragraph removal. All other changes might be different from site to site and it's better to leave them out of the PR.

File: config/default/field.field.node.hs_private_page.field_hs_priv_page_components.yml [hunk 5]
```diff
@@ -28,79 +28,44 @@ settings:
...
-        weight: -39
-        enabled: true
-      hs_timeline_item:
-        weight: -25
+        weight: 14
         enabled: false
       hs_view:
-        weight: -34
-        enabled: false
-      hs_webform:
```
Comment (cienvaras) [inline config/default/field.field.node.hs_private_page.field_hs_priv_page_components.yml:97]: In overridden config, try to include only the lines necessary for the the `hs_webform` paragraph removal. All other changes might be different from site to site and it's better to leave them out of the PR.

**Review by cienvaras — CHANGES_REQUESTED** [review]
@codechefmarc Looks good, just some minor details to fix, please check the inline comments.

There's an error in the codeception tests, it seems that there's an extra item in the shortcuts menu. It doesn't seem to be related to your changes, but please take a look if you have time available.

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 2 above)
Comment (codechefmarc) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: It correctly gets uninstalled via the `core.extension` but the others needed to be done like this before the config gets imported. So, I can add here, but it's not necessary.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-colorful-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-traditional-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0

Visual Diffs:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-colorful-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-traditional-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0

Visual Diffs:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0/visualdiffs

File: config/default/user.role.authenticated.yml (hunk 1 above)
Comment (codechefmarc) [reply config/default/user.role.authenticated.yml]: Removed.

File: config/default/field.field.node.hs_basic_page.field_hs_page_components.yml (hunk 3 above)
Comment (codechefmarc) [reply config/default/field.field.node.hs_basic_page.field_hs_page_components.yml:97]: Removed other changes.

File: config/default/field.field.node.hs_basic_page.field_hs_page_hero.yml (hunk 4 above)
Comment (codechefmarc) [reply config/default/field.field.node.hs_basic_page.field_hs_page_hero.yml:95]: Removed other changes.

File: config/default/field.field.node.hs_private_page.field_hs_priv_page_components.yml (hunk 5 above)
Comment (codechefmarc) [reply config/default/field.field.node.hs_private_page.field_hs_priv_page_components.yml:97]: Removed other changes.

**Comment by codechefmarc** [comment]
@cienvaras - Config changes made. Also removed a default content of adding a "Manage webforms" shortcut to new sites - that seemed to initially clear up the codeception test, but then there were other errors about adding a course page and then again shortcuts as well. So not sure. If you find anything, please let me know. Thanks!

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-colorful-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-traditional-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0

Visual Diffs:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0/visualdiffs

**Review by cienvaras — APPROVED** [review]
@codechefmarc Thanks for the fixes! Tests are passing now, I just had to update the test to match the new number of shortcuts after removing the webforms one.

@ahughes3 It's ready for your review.

**Review by ahughes3 — APPROVED** [review]
looks good!

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-colorful-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-traditional-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0

Visual Diffs:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0/visualdiffs

**Comment by joegl** [comment]
👀 I took a quick look at this one and will review it fully later. I'm excited to get this module deprecated 😄 
@ahughes3 Let me know if you want this one bumped in priority.

**Comment by ahughes3** [comment]
@joegl because it's not a forward facing change and we have a follow up to remove the module I'm for bumping up priority to get it into 11.2.5

**Comment by joegl** [comment]
> @joegl because it's not a forward facing change and we have a follow up to remove the module I'm for bumping up priority to get it into 11.2.5

I'm about to deploy 11.2.5 now but I'll bump it up for the next release.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-colorful-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com
- https://hs-traditional-bbygkd8loksxzrmo2pvpvdwrfp4xdzhs.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0

Visual Diffs:
- https://dashboard.tugboatqa.com/66ea1c55faaeb91b536e5cf0/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-k5s2eaz0xfyfhwwu2xniw1nwyrwm5f6v.tugboatqa.com
- https://hs-colorful-k5s2eaz0xfyfhwwu2xniw1nwyrwm5f6v.tugboatqa.com
- https://hs-traditional-k5s2eaz0xfyfhwwu2xniw1nwyrwm5f6v.tugboatqa.com
- https://english-k5s2eaz0xfyfhwwu2xniw1nwyrwm5f6v.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67118eb8213f2908415f8206

Visual Diffs:
- https://dashboard.tugboatqa.com/67118eb8213f2908415f8206/visualdiffs

**Comment by cienvaras** [comment]
@joegl I just noticed that some of the feedback you gave us for #1643 also applies here. I'll add the fixes and let you know when it's ready for review again.

**Comment by cienvaras** [comment]
@joegl PR updated and ready to review again. Thanks!

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-k5s2eaz0xfyfhwwu2xniw1nwyrwm5f6v.tugboatqa.com
- https://hs-colorful-k5s2eaz0xfyfhwwu2xniw1nwyrwm5f6v.tugboatqa.com
- https://hs-traditional-k5s2eaz0xfyfhwwu2xniw1nwyrwm5f6v.tugboatqa.com
- https://english-k5s2eaz0xfyfhwwu2xniw1nwyrwm5f6v.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67118eb8213f2908415f8206

Visual Diffs:
- https://dashboard.tugboatqa.com/67118eb8213f2908415f8206/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-3dueql06rbx7qauq5t7gvd21asvjklfq.tugboatqa.com
- https://hs-colorful-3dueql06rbx7qauq5t7gvd21asvjklfq.tugboatqa.com
- https://hs-traditional-3dueql06rbx7qauq5t7gvd21asvjklfq.tugboatqa.com
- https://english-3dueql06rbx7qauq5t7gvd21asvjklfq.tugboatqa.com
- https://history-3dueql06rbx7qauq5t7gvd21asvjklfq.tugboatqa.com
- https://west-3dueql06rbx7qauq5t7gvd21asvjklfq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672a882c21db79eb23217a54

**Comment by ahughes3** [comment]
@cienvaras @codechefmarc does the branch conflicts need to be resolved before SWS can review again?

**Comment by cienvaras** [comment]
@ahughes3 Done, conflicts fixed.

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install [hunk 6]
```diff
@@ -1026,3 +1028,113 @@ function su_humsci_profile_update_9714() {
...
 }
+
+/**
+ * Remove any existing hs_webform paragraph entities.
+ */
+function su_humsci_profile_update_9715($sandbox) {
+  $paragraph_storage = \Drupal::entityTypeManager()->getStorage('paragraph');
+  $paragraph_machine_name = 'hs_webform';
+
+  if (empty($sandbox['ids'])) {
```
Comment (pookmish) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install:1049]: I doubt there's more than a dozen on any given site, but it's good practice to use the sandbox when making entity changes that might be of an unknown quantity. :+1:

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install [hunk 7]
```diff
@@ -1026,3 +1028,113 @@ function su_humsci_profile_update_9714() {
...
+ * Delete hs_webform paragraph type and remove webform config.
+ */
+function su_humsci_profile_update_9716() {
+  $paragraph_machine_name = 'hs_webform';
+  // Delete the field and field storage on the paragraph.
+  $fields = \Drupal::entityTypeManager()
+    ->getStorage('field_config')
+    ->loadByProperties(['bundle' => $paragraph_machine_name]);
+
+  if ($fields) {
```
Comment (pookmish) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: You don't need to check for this. If it's an empty array, the foreach will just be skipped. 

But the fun fact is that you can delete the field storage and it'll automatically delete any field config associated to it.

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install [hunk 8]
```diff
@@ -1026,3 +1028,113 @@ function su_humsci_profile_update_9714() {
...
+      // Delete the field configuration (instance) from the paragraph type.
+      $field->delete();
+    }
+  }
+
+  // Delete the field_hs_webform field storage.
+  $field_storage = FieldStorageConfig::load('field_hs_webform');
+  if ($field_storage) {
+    $field_storage->delete();
+  }
```
Comment (pookmish) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Not really necessary, but something you could do to make it more simple:
```suggestion
  FieldStorageConfig::load('field_hs_webform')?->delete();
```

Same with the other lines below.

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install [hunk 9]
```diff
@@ -1026,3 +1028,113 @@ function su_humsci_profile_update_9714() {
...
+    }
+  }
+
+  // Remove webform existing config - needs to be done before uninstall.
+  $query = \Drupal::database()
+    ->select('config', 'c')
+    ->fields('c', ['name'])
+    ->condition('name', '%' . Database::getConnection()->escapeLike('webform') . '%', 'LIKE')
+    ->execute();
+  $config_names = $query->fetchCol();
```
Comment (pookmish) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Use `\Drupal::configFactory()->listAll('webform.')` instead of db query.

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install [hunk 10]
```diff
@@ -1026,3 +1028,113 @@ function su_humsci_profile_update_9714() {
...
+    'node.hs_private_page.field_hs_priv_page_components' => 'node.field_hs_priv_page_components',
+  ];
+
+  /** @var \Drupal\Core\Entity\EntityStorageInterface $field_config_storage */
+  $field_config_storage = \Drupal::service('entity_type.manager')->getStorage('field_config');
+
+  /** @var \Drupal\Core\Entity\EntityStorageInterface $fieldConfigStorage */
+  $field_storage = \Drupal::service('entity_type.manager')->getStorage('field_storage_config');
+
+  foreach ($configs_to_change as $field_config => $field_storage_config) {
```
Comment (pookmish) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Is this necessary if the configs are to be synced after the db update?

**Review by pookmish — CHANGES_REQUESTED** [review]
Configs all look fine, mainly just the one issue that needs to be replaced.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://hs-colorful-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://hs-traditional-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://english-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://history-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://west-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674738b5d871e5f0740cca7f

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 7 above)
Comment (cienvaras) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Thanks for the tip! I removed this section because we're removing the storage below.

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 8 above)
Comment (cienvaras) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Fixed!

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 9 above)
Comment (cienvaras) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: I checked, and the the list of config items is a bit different when using `\Drupal::configFactory()->listAll('webform.')`, because we only get items that start with `webform`, while the db query includes items that include `webform` anywhere in the name:

- `editor.editor.webform_default`
- `filter.format.webform_default`
- `system.action.webform_archive_action`
- `system.action.webform_close_action`
- `system.action.webform_delete_action`
- `system.action.webform_open_action`
- `system.action.webform_submission_delete_action`
- `system.action.webform_submission_make_lock_action`
- `system.action.webform_submission_make_sticky_action`
- `system.action.webform_submission_make_unlock_action`
- `system.action.webform_submission_make_unsticky_action`
- `system.action.webform_unarchive_action`
- `ultimate_cron.job.webform_cron`

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 10 above)
Comment (cienvaras) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Config for `field.field.node.hs_basic_page.field_hs_page_components` and `field.field.node.hs_private_page.field_hs_priv_page_components` is [partially overridden](https://github.com/SU-HSDO/suhumsci/blob/416ecc8ccfedb3e759e7c91e129a92b8acd7e363/config/default/config_ignore.settings.yml#L23), so it won't be synced when config is imported. The config for `field_hs_collection_items` is not overridden, so I removed it, but noticed that `field.field.node.hs_basic_page.field_hs_page_hero` was missing, so I added it.

**Comment by cienvaras** [comment]
@pookmish Thanks for the feedback! I fixed some of the issues and left comments in the other cases. Please let me know if any other fix is needed. (cc. @ahughes3)

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 9 above)
Comment (pookmish) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Ah I see. Maybe just do `->listAll()` and then you can `array_filter` the config names that include webform. I'm just not liking a db query directly like this when there are other ways.

Comment (cienvaras) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: @pookmish Yes, that makes sense. It's fixed now, thanks.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://hs-colorful-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://hs-traditional-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://english-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://history-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://west-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674738b5d871e5f0740cca7f

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install [hunk 11]
```diff
@@ -1026,3 +1027,90 @@ function su_humsci_profile_update_9714() {
...
+  $config_names = array_filter($config_factory->listAll(), function ($item) {
+    return str_contains($item, 'webform');
+  });
+
+  foreach ($config_names as $config) {
+    $config_factory->getEditable($config)->delete();
+    $message = t('Deleted @config', [
+      '@config' => $config,
+    ]);
+    Drush::logger()->notice($message);
```
Comment (pookmish) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Replace with normal logger or messenger. Drush shoudn't be used in update hooks like this.

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install (hunk 10 above)
Comment (pookmish) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install]: Yes, they are partially overridden for a reason. Doing this, re-syncs the configuration with the file system, essentially ignoring why that is partially ignored. This can have negative results.

**Review by pookmish — CHANGES_REQUESTED** [review]

**Comment by pookmish** [comment]
Also, there are merge conflicts that need to be resolved.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1625-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://hs-colorful-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://hs-traditional-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://english-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://history-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com
- https://west-0aksq5zmygcfebeqxg5uun9wxad8fwrw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674738b5d871e5f0740cca7f

**Comment by cienvaras** [comment]
@pookmish Thanks! Both issues and the merge conflicts are fixed now.

**Review by pookmish — APPROVED** [review]
